src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/controller/CertificateController.kt

```diff
@@ -19,18 +19,20 @@
 
 package org.eclipse.tractusx.bpdmcertificatemanagement.controller
 
-import org.springframework.http.MediaType
-import org.springframework.web.bind.annotation.GetMapping
-import org.springframework.web.bind.annotation.PathVariable
-import org.springframework.web.bind.annotation.RequestMapping
+import org.eclipse.tractusx.bpdmcertificatemanagement.dto.request.CertificateDocumentRequestDto
+import org.eclipse.tractusx.bpdmcertificatemanagement.dto.response.CertificateDocumentResponseDto
+import org.eclipse.tractusx.bpdmcertificatemanagement.service.CertificateService
+import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.RestController
 
 @RestController
-@RequestMapping("api/catena/certificate", produces = [MediaType.APPLICATION_JSON_VALUE])
-class CertificateController {
+class CertificateController(
+    val certificateService: CertificateService
+): CertificateApi {
+
+    override fun setCertificateDocument(certificateDocumentRequestDto: CertificateDocumentRequestDto): ResponseEntity<CertificateDocumentResponseDto> {
+        val result = certificateService.createCertificate(certificateDocumentRequestDto)
+        return ResponseEntity.ok(result)
```

Comment:
Or you can use the created response code 201

Reply:
Yeah 201 and 422 status code adapted in next pull request i shall update unique key while creating certificate response.